plugins/ledger/ledger.py

```diff
@@ -176,17 +177,8 @@ def perform_hw1_preflight(self):
                 pin = pin.encode()
                 self.dongleObject.verifyPin(pin)
 
-            try:
-                try:
-                    self.dongleObject.getWalletPublicKey("44'/145'/0'/0/0", showOnScreen=False, cashAddr=True)
-                except BTChipException as e:
-                    # This call fails on HW1/Nano with a specific error, due to it not supporting cashaddr
-                    if (e.sw != 0x6b00):
-                        raise e
-                self.cashaddrSWSupported = True
-            except TypeError:
-                self.cashaddrSWSupported = False
-
+            gwpkArgSpecs = inspect.getfullargspec(self.dongleObject.getWalletPublicKey)
+            self.cashaddrSWSupported = 'cashAddr' in gwpkArgSpecs.args
```

Comment:
Yep. I think this is cleaner... as long as you say it's a good test (ie the API doesn't give you a dongleObject that accepts the key if the device does not)...